changelog/unreleased/issue-13888.toml

```diff
@@ -0,0 +1,7 @@
+# Entry type according to https://keepachangelog.com/en/1.0.0/
+# One of: a(dded), c(hanged), d(eprecated), r(emoved), f(ixed), s(ecurity)
+type = "fixed"
+message = "Pipeline function flatten_json respects the original JSON types. An optional parameter is provided for back-compat."
```

Comment:
```suggestion
message = "The pipeline function flatten_json now respects the original JSON types. An optional parameter is provided for backwards compatibility"
```

Reply:
I guess we should note that also in `UPGRADING.md` because it's a breaking change.
I do agree that we should change the default though, it's better to do it now, rather than to have the wrong default forever in our product.

Reply:
fixed

Reply:
Oops, missed that - fixed it